opentracing-util/src/main/java/io/opentracing/util/GlobalTracer.java

```diff
@@ -84,43 +82,73 @@ public static Tracer get() {
         return INSTANCE;
     }
 
+    /**
+     * Identify whether a {@link Tracer} has previously been registered.
+     * <p>
+     * This check is useful in scenarios where more than one component may be responsible
+     * for registering a tracer. For example, when using a Java Agent, it will need to determine
+     * if the application has already registered a tracer, and if not attempt to resolve and
+     * register one itself.
+     *
+     * @return Whether a tracer has been registered
+     */
+    public static synchronized boolean isRegistered() {
+        return !(GlobalTracer.tracer instanceof NoopTracer);
+    }
+
     /**
      * Register a {@link Tracer} to back the behaviour of the {@link #get() global tracer}.
      * <p>
-     * Registration is a one-time operation, attempting to call it more often will result in a runtime exception.
+     * The tracer is provided through a {@linkplain Callable} that will only be called if the global tracer is absent.
+     * Registration is a one-time operation. Once a tracer has been registered, all attempts at re-registering
+     * will return {@code false}.
      * <p>
      * Every application intending to use the global tracer is responsible for registering it once
      * during its initialization.
      *
-     * @param tracer Tracer to use as global tracer.
-     * @throws RuntimeException if there is already a current tracer registered
+     * @param provider Provider for the tracer to use as global tracer.
+     * @return {@code true} if the provided tracer was registered as a result of this call,
+     * {@code false} otherwise.
+     * @throws NullPointerException  if the tracer provider is {@code null} or provides a {@code null} Tracer.
+     * @throws RuntimeException      any exception thrown by the provider gets rethrown,
+     *                               checked exceptions will be wrapped into appropriate runtime exceptions.
      */
-    public static synchronized void register(final Tracer tracer) {
-        if (tracer == null) {
-            throw new NullPointerException("Cannot register GlobalTracer <null>.");
-        }
-        if (tracer instanceof GlobalTracer) {
-            LOGGER.log(Level.FINE, "Attempted to register the GlobalTracer as delegate of itself.");
-            return; // no-op
-        }
-        if (isRegistered() && !GlobalTracer.tracer.equals(tracer)) {
-            throw new IllegalStateException("There is already a current global Tracer registered.");
+    public static synchronized boolean registerIfAbsent(final Callable<Tracer> provider) {
+        requireNonNull(provider, "Cannot register GlobalTracer from provider <null>.");
+        if (!isRegistered()) try {
```

Comment:
This is not a common style in Java, even it is legal
```
 if (!isRegistered()) try {
```
Please consider
```
 if (!isRegistered()) {
     try {
     }
}
```

Reply:
Personally (matter of taste) I find the intention better communicated to keep " if _condition_ try " on a single line.

But I'll change it. Merging the deprecation soon is more important for me.

Reply:
You could keep your habit :)  It is a suggestion only.

Reply:
@wu-sheng I will in my own projects. 
Since this is a public API however, I agree that any uncommon style should be avoided when possible. :+1: